File: src/backend_dotnet/HRMApi/Models/Request.cs
using System;
using System.Collections.Generic;

namespace HRMApi.Models;

public partial class Request
{
    public int Id { get; set; }

    public int EmployeeId { get; set; }

    public string Description { get; set; } = null!;

    public DateTime? StartTime { get; set; }

    public DateTime? EndTime { get; set; }

    public string Type { get; set; } = null!;

    public string? Attachment { get; set; }

    public string? Status { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public virtual ICollection<ApprovalHistory> ApprovalHistories { get; set; } = new List<ApprovalHistory>();

    public virtual Employee Employee { get; set; } = null!;
}

File: src/backend_java/src/main/java/com/group07/human_resource_management/entity/Request.java
package com.group07.human_resource_management.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "request")
@Getter @Setter
@Builder @NoArgsConstructor @AllArgsConstructor
public class Request {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // employee_id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employee_id", nullable = false)
    private Employee employee;

    private String description;

    @Column(name = "start_time")
    private LocalDateTime startTime;

    @Column(name = "end_time")
    private LocalDateTime endTime;

    private String type; // wfh, leave, overtime, …

    private String attachment;

    private String status;

    private Double duration;

    @Column(name = "leave_mode")
    private String leaveMode; // DAY, HALF_DAY, SHORT_HOUR

    private String session; // MORNING, AFTERNOON

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
